### ISSUE TITLE
Spanner: implement optimizer stats option

### ISSUE THREAD

**Issue by devjgm** [googleapis/google-cloud-cpp#4820]
b/162993810

**Comment by devjgm** [comment]
Here's the draft PR for this change: https://github.com/googleapis/google-cloud-cpp/compare/master...devjgm:optimizer_statistics_package?expand=1

It cannot be submitted yet because it depends on this branch github.com/googleapis/googleapis/compare/spanner_query_stats_package which has not yet been merged into master. Once this is merged into master, we should be able to merge this feature.

**Comment by devjgm** [comment]
This issue moved to Q4 due to product needs.

**Comment by devjgm** [comment]
This one is done, right? @devbww

**Comment by devbww** [comment]
"Done" in that (1) the protos are published and (2) we have a pending PR (mostly from you), which I am keeping up to date.

But "not done" in that the backend currently doesn't support the `SPANNER_SYS.OPTIMIZER_STATISTICS_PACKAGES` table and who-knows-what-else about the feature.  Last word in the internal "client features spec" was "A bunch of backend changes were required but they said it would be ready to ship in mid Q1 2021."

I'll try to get some clarity in the next client-sync meeting.

Aside: Some of the confusion about what is done and what is not, I think, is due to the blurred difference between `ExecuteSqlRequest.QueryOptions.optimizer_version`, which we have released, and `ExecuteSqlRequest.QueryOptions.optimizer_statistics_package`, which is what we're talking about here.
